Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -4179,6 +4178,31 @@ public int GetPipPriority(PipId pipId)
         /// <inheritdoc />
         public DirectoryTranslator DirectoryTranslator { get; }
 
+        [SuppressMessage("Microsoft.Design", "CA1033:InterfaceMethodsShouldBeCallableByChildTypes")]
+        string IPipExecutionEnvironment.GetProducerExecutionInfo(in FileOrDirectoryArtifact artifact)
+        {
+            var producerId = PipGraph.GetProducer(artifact);
+            if (!producerId.IsValid)
+            {
+                return "N/A";
+            }
+
+            var producer = PipGraph.GetPipFromPipId(producerId);
+
+            RunnablePipPerformanceInfo perfInfo = m_runnablePipPerformance[producerId];
+
+            PipExecutionStep step = PipExecutionStep.ExecuteProcess;
+            if (perfInfo.StepDurations[(int)step].TotalMilliseconds == 0)

```

Comment:
>(int)step].TotalMilliseconds == 0 [](start = 39, length = 33)

This feels like a fuzzy way to get this signal. Can you utilize some other state like whether CacheLookupPerfInfo was set?